app/controllers/application_controller.rb

```diff
@@ -35,6 +35,12 @@ def switch_locale(&action)
     end
   end
 
+  # Overrides Devise::Controller::StoreLocation.store_location_for to check if
+  # URL is too long to store in the session
+  def store_location_for(resource_or_scope, location)
+    super unless location && location.length > 150
```

Comment:
What about setting location=request.host if location > OUR_MAX_LIMIT. So that to avoid potential infinite loop and unexpected behaviour?

Reply:
I think devise automatically redirects to the home page if there is nothing in session?

Reply:
But what if there are some from previous requests which aren't cleared yet ?